drivers/at24cxxx/Kconfig

```diff
@@ -5,64 +5,135 @@
 # directory for more details.
 #
 
-choice
-    bool "AT24CXXX EEPROM"
-    depends on HAS_PERIPH_I2C
+menuconfig MODULE_AT24CXXX
+    bool
```

Comment:
Optional would actually not select anything automatically (which is what we want with the `HAVE_*`).

My alternative solution would be to add a default invisible choice of `MINIMAL` as indicated with the else comment in the `at24cxxx_defines.h`.  Providing one's own params would overwrite this anyways (which is what would be needed for the multiple variants).

Therefore, if someone wants to define a board that has multiple eeproms they would have to manually enter the eeprom params into the board configuration and select `HAVE_AT24CXXX` to get the plain module.

Does that sound like a nice solution?

Reply:
> Optional would actually not select anything automatically (which is what we want with the HAVE_*).

Ok

> My alternative solution would be to add a default invisible choice of MINIMAL

This would also be good IMO. I would prefer MULTIPLE.
But what do you mean with invisible? As a user I think I would like to see what is selected.

Should there be a `default AT24CXXX_MINIMAL` in `choice AT24CXXX_VARIANT` for the case that no `HAVE_*` is provided? Because if I remove `CONFIG_MODULE_AT24C256=y`, and then do 
`TEST_KCONFIG=1 BOARD=nucleo-f767zi make menuconfig` in `tests/drivers_at24cxxx`, the `AT24MAC` is selected by default, which may appear a bit misleading.

> Therefore, if someone wants to define a board that has multiple eeproms they would have to manually enter the eeprom params into the board configuration and select HAVE_AT24CXXX to get the plain module.

Yeah in the end this takes precedence.

Reply:
> But what do you mean with invisible?

Hmm I don't know why I said that, it is selectable and _should _ be selected as default when selecting the `MODULE_AT24CXXX` if no `HAVE_*` are entered

Reply:
>  if I remove CONFIG_MODULE_AT24C256=y, and then do
TEST_KCONFIG=1 BOARD=nucleo-f767zi make menuconfig in tests/drivers_at24cxxx, the AT24MAC is selected by default, which may appear a bit misleading.

Super misleading, I will look into that...

Reply:
I pushed a fix, what happened was:

- in the `drivers/at24cxxx/Kconfig`, if there are no `HAVE_*` to select anything it will default to the first in the list (now named `AT24CXXX_MULTIPLE` based it being the first in the choice position.
- the `drivers/at24mac/Kconfig` adds `MODULE_AT24MAC_EEPROM` via overloading the `AT24CXXX_VARIANT`.
- Since `MODULE_AT24MAC_EEPROM` was the first in this choice, I guess the default just used this.
- To fix I specify a default if there is no `HAVE_*`

I tested with `TEST_KCONFIG=1 BOARD=nucleo-f767zi make info-modules -C tests/driver_at24cxxx/ | grep at` after removing the `CONFIG_MODULE_AT24C256`.

Reply:
Yup, didn't even try it like that.  Both suggestions added.